src/Data/Models/FMUTXO.cs

```diff
@@ -29,14 +34,31 @@ public class FMUTXO : Entity, IEquatable<FMUTXO>
         public uint OutputIndex { get; set; }
 
         public long SatsAmount { get; set; }
-
+        
+        public string? Address { get; set; }
+        
+        public bool IsFrozen { get; set; }
+        
+        public List<UTXOTag>? Tags { get; set; }
+        
         #region Relationships
 
+        // M-N Because if the UTXO is used in a request that gets cancelled,
+        // the UTXO should be unlocked and assigned to another request
         public List<ChannelOperationRequest> ChannelOperationRequests { get; set; }
 
+        // Idem as ChannelOperationRequests
         public List<WalletWithdrawalRequest> WalletWithdrawalRequests { get; set; }
+        
+        public int WalletId { get; set; }
+        public Wallet Wallet { get; set; }
 
         #endregion Relationships
+        
+        [NotMapped]
+        public bool IsSpent => (!ChannelOperationRequests.IsNullOrEmpty() || !WalletWithdrawalRequests.IsNullOrEmpty())
+                               && (ChannelOperationRequests.Any(x => x.Status == ChannelOperationRequestStatus.OnChainConfirmed) || 
+                                   WalletWithdrawalRequests.Any(x => x.Status == WalletWithdrawalRequestStatus.OnChainConfirmed));
```

Comment:
What if it was spent outside of nodeguard?

Reply:
Good call; I don't think it's needed, tho. Will remove it.